src/sonic-config-engine/portconfig.py

```diff
@@ -51,4 +120,94 @@ def parse_port_config_file(port_config_file):
+        ports[intf_name]['lanes'] = ','.join(lanes.split(",")[alias_start:alias_start+step])
+        if speed:
+            ports[intf_name]['speed'] = speed
+        else:
+            raise Exception('Regex return for speed is None...')
+        ports[intf_name]['index'] = index.split(",")[alias_start]
+        ports[intf_name]['admin_status'] = "up"
+
+        parent_intf_id += step
+        alias_start += step
 
+
+def parse_platform_json_file(port_config_file, interface_name=None, target_brkout_mode=None):
+    ports = {}
+    port_alias_map = {}
+
+    # Read 'platform.json' file
+    try:
+        with open(port_config_file) as fp:
+            try:
+                data = json.load(fp)
+            except json.JSONDecodeError as e:
+                raise Exception("JSONDecodeError:", e)
+        global port_dict
+        port_dict = ast.literal_eval(json.dumps(data))
+    except:
+        print("error occurred while parsing json:", sys.exc_info()[1])
+
+    for intf in port_dict:
+        if str(interface_name) == intf:
+            brkout_mode = target_brkout_mode
+        else:
+            brkout_mode = port_dict[intf][BRKOUT_MODE]
+        index = port_dict[intf]['index']
+        alias_at_lanes = port_dict[intf]['alias_at_lanes']
+        lanes = port_dict[intf]['lanes']
+
+        # if User does not specify brkout_mode, take default_brkout_mode from platform.json
+        if brkout_mode is None:
+            brkout_mode = port_dict[intf][BRKOUT_MODE]
+
+        # Get match_list for Asymmetric breakout mode
+        if re.search("\+",brkout_mode) is not None:
+            brkout_parts = brkout_mode.split("+")
+            match_list = [re.match(BRKOUT_PATTERN, i).groups() for i in brkout_parts]
+
+        # Get match_list for Symmetric breakout mode
+        else:
+            match_list = [re.match(BRKOUT_PATTERN, brkout_mode).groups()]
+
+        """
+        Example of match_list for some breakout_mode using regex
+            Breakout Mode -------> Match_list
+            -----------------------------
+            2x25G(2)+1x50G(2) ---> [('2', '25G', None, '(2)', '2'), ('1', '50G', None, '(2)', '2')]
+            1x50G(2)+2x25G(2) ---> [('1', '50G', None, '(2)', '2'), ('2', '25G', None, '(2)', '2')]
+            1x100G[40G] ---------> [('1', '100G', '[40G]', None, None)]
+            2x50G ---------------> [('2', '50G', None, None, None)]
+        """
+        if match_list is not None:
+            offset = 0
+            parent_intf_id = int(re.search("Ethernet(\d+)", intf).group(1))
+            for k in match_list:
+                num_lane_used, speed, alt_speed, _ , assigned_lane = k[0], k[1], k[2], k[3], k[4]
+
+                # In case of symmetric mode
+                if assigned_lane is None:
+                    assigned_lane = len(lanes.split(","))
+
+                gen_port_config(ports, parent_intf_id, index, alias_at_lanes, lanes,num_lane_used, speed, assigned_lane,  offset)
```

Comment:
I was thinking to pass k  to gen_port_config(), instead of passing 9 parameters to a function

Reply:
modified as requested.